Sources/Fuzzilli/FuzzIL/ProgramTypes.swift

```diff
@@ -73,7 +69,63 @@ public struct ProgramTypes: Equatable, Sequence {
         return types[variable]?.last(where: { $0.index <= instrIndex })?.type ?? .unknown
     }
 
-    public func makeIterator() -> VariableMap<[TypeIndexPair]>.Iterator {
+    // Filter out only runtime types
+    public func getRuntimeTypes() -> ProgramTypes {
+        var runtimeTypes = ProgramTypes()
+        for (variable, instrTypes) in types {
+            for typeInfo in instrTypes {
+                guard typeInfo.quality == .runtime else { continue }
+
+                runtimeTypes.setType(
+                    of: variable, to: typeInfo.type, at: typeInfo.index, quality: .runtime
+                )
+            }
+        }
+        return runtimeTypes
+    }
+
+    // Format ProgramTypes struct so searching for type changes at instruction is easier
+    public func getInvertedStructure(for program: Program) -> [[(Variable, Type)]] {
+        var typesMap: [[(Variable, Type)]] = Array(repeating: [], count: program.size)
+        for (variable, instrTypes) in types {
+            for typeInfo in instrTypes {
+                typesMap[typeInfo.index].append((variable, typeInfo.type))
+            }
+        }
+        return typesMap
+    }
+
+    public func makeIterator() -> VariableMap<[TypeInfo]>.Iterator {
         return types.makeIterator()
     }
 }
+
+extension ProgramTypes: ProtobufConvertible {
+    public typealias ProtobufType = [UInt32: Fuzzilli_Protobuf_InstrTypes]
+
+    func asProtobuf() -> ProtobufType {
+        var protobuf = ProtobufType()
+        for (variable, instrTypes) in self {
+            protobuf[UInt32(variable.number)] = Fuzzilli_Protobuf_InstrTypes()
+            for typeInfo in instrTypes {
+                protobuf[UInt32(variable.number)]!.typeInfo.append(typeInfo.asProtobuf())
```

Comment:
Nit: how about you keep the `Fuzzilli_Protobuf_InstrTypes()` in a local variable at first, append all of these, then attach it to `protobuf[UInt32(variable.number)]`?